### ISSUE TITLE
Has anyone ever encountered the problem that deleting a light will quit the software?

### ISSUE THREAD

**Issue by Owiilnice** [leomoon-studios/blender-light-studio#78]
The version I'm using is blender 4.02
Create a light and then click Delete to exit the software

**Comment by leomoon** [comment]
I'll do some tests to see if I can replicate it.

Do you remember how many profiles/lights you had?

**Comment by Owiilnice** [comment]
This issue also occurs with newly created files
&nbsp;


775908105
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Amin ***@***.***&gt;; 
发送时间: 2023年12月27日(星期三) 上午10:23
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [leomoon-studios/leomoon-lightstudio] Has anyone ever encountered the problem that deleting a light will quit the software? (Issue #78)





 
I'll do some tests to see if I can replicate it.
 
Do you remember how many profiles/lights you had?

**Comment by greasemonkey2603** [comment]
Yes im indeed having the same issue even in a newly created file.
And the addon is showing this in the preferences tab.
![image](https://github.com/leomoon-studios/leomoon-lightstudio/assets/53222223/8eb1f075-6e57-4037-99e3-1747578cbb8d)

**Comment by leomoon** [comment]
> Yes im indeed having the same issue even in a newly created file. And the addon is showing this in the preferences tab. ![image](https://private-user-images.githubusercontent.com/53222223/293455407-8eb1f075-6e57-4037-99e3-1747578cbb8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM5MzI4NTQsIm5iZiI6MTcwMzkzMjU1NCwicGF0aCI6Ii81MzIyMjIyMy8yOTM0NTU0MDctOGViMWYwNzUtNmU1Ny00MDM3LTk5ZTMtMTc0NzU3OGNiYjhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMwVDEwMzU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OTkwNDM3MjVmNTlhNDkzNTE0NjNlYWIwMDJlNzc2ZDM5ZDE0MTdjMzg0ZDY5OGUxMjM4ZTgwOGZhMzYwMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tl4BYwIglmbjfX0_vMn54eKBJNhYCUU1Cms42Ugbi5w)

Only stable versions of Blender 4.0 are supported.

- What is your OS?
- What is the Blender version?

**Comment by greasemonkey2603** [comment]
> > Yes im indeed having the same issue even in a newly created file. And the addon is showing this in the preferences tab. ![image](https://private-user-images.githubusercontent.com/53222223/293455407-8eb1f075-6e57-4037-99e3-1747578cbb8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM5MzI4NTQsIm5iZiI6MTcwMzkzMjU1NCwicGF0aCI6Ii81MzIyMjIyMy8yOTM0NTU0MDctOGViMWYwNzUtNmU1Ny00MDM3LTk5ZTMtMTc0NzU3OGNiYjhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMwVDEwMzU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OTkwNDM3MjVmNTlhNDkzNTE0NjNlYWIwMDJlNzc2ZDM5ZDE0MTdjMzg0ZDY5OGUxMjM4ZTgwOGZhMzYwMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tl4BYwIglmbjfX0_vMn54eKBJNhYCUU1Cms42Ugbi5w)
> 
> Only stable versions of Blender 4.0 are supported.
> 
> * What is your OS?
> * What is the Blender version?

Sorry for not specifying earlier
OS: windows 11
Blender version: 4.0.2

**Comment by leomoon** [comment]
I haven't been able to replicate this.

Could you please see if you can send me Blender crash logs right after a crash? @greasemonkey2603 @Owiilnice 
https://docs.blender.org/manual/en/latest/troubleshooting/crash.html#windows

Blender console output can be helpful also (when performing the tasks that could crash Blender): `Blender > Window > Toggle System Console`

**Comment by greasemonkey2603** [comment]
Here is the crash log of a fresh scene with only a cube and a plane and the lightstudio setup.

[blender.crash.txt](https://github.com/leomoon-studios/leomoon-lightstudio/files/13799160/blender.crash.txt)

**Comment by leomoon** [comment]
Thanks. The crash logs shows `MESHmachine` then `light-studio` after that. They might be conflicting with each other.

```
# Python backtrace
  File "C:\Users\reyya\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\MESHmachine\handlers.py", line 70 in stashes_HUD
  File "C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py", line 68 in _view_layer_update
  File "C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py", line 112 in __call__
  File "C:\Users\reyya\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\leomoon-lightstudio-master\light_operators.py", line 592 in execute
  File "C:\Users\reyya\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\leomoon-lightstudio-master\light_operators.py", line 606 in invoke
```

Could you please disable `MESHmachine` and see if `light-studio` still crashes?

**Comment by leomoon** [comment]
I think we have some clues what is going on. We're working on a fix.

Thanks for your help. Any other info is welcome.

**Comment by Owiilnice** [comment]
Discovered via blender.crash.txt
Plugins conflict with MACHIN3 or CameraHelper

**Comment by leomoon** [comment]
I did more testing.

Plugin works fine if it is just installed.

As soon as Blender is restarted, plugin stops working.

**Comment by greasemonkey2603** [comment]
@leomoon any updates regarding the issue?

**Comment by henryTPE** [comment]
when open control panel, blender also crash.
OS: windows 11
Blender version: 4.0.2
<img width="673" alt="螢幕擷取畫面 2024-01-11 123038" src="https://github.com/leomoon-studios/leomoon-lightstudio/assets/31567877/91a95f7f-fa76-4c44-b0bc-aec132c5f5cd">

**Comment by vaibhavkulkarni5w1h** [comment]
I am also having same issue. the blender is crashing when I am trying to open control panel

**Comment by vaibhavkulkarni5w1h** [comment]
can someone please help me on how to solve this

**Comment by leomoon** [comment]
Hi,

There's no solution to fix this yet. There seems to be some undocumented Blender API change.

We're still working on it.

Sorry for any inconvenience and thank you for your patience. :)

**Comment by Owiilnice** [comment]
> can someone please help me on how to solve this有人能帮我解决这个问题吗

At the moment, you can only try to close all of your plugins to get it working

**Comment by jksnkjnd** [comment]
Has the plug-in crash issue been resolved? The flowers have faded after waiting. If it has been solved, please give the author a download link. I will be very grateful. Because I can't find the new download link!

**Comment by leomoon** [comment]
Thanks for your patience.

We have found a solution and we're testing it.

We will release a new version in a week or so.

**Comment by jksnkjnd** [comment]
> Thanks for your patience.
> 
> We have found a solution and we're testing it.
> 
> We will release a new version in a week or so.

That's great, looking forward to it

**Comment by leomoon** [comment]
this should be fixed in LightStudio 2.15.1.

Please make sure the old LightStudio version is removed.

**Comment by jksnkjnd** [comment]
It finally lived up to expectations, thank you very much
